internal/stats/instrumentregistry/instrument_registry.go

```diff
@@ -0,0 +1,199 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package instrumentregistry is a instrument registry that gRPC components can
+// use to register instruments (metrics).
+package instrumentregistry
+
+import (
+	"log"
+
+	"google.golang.org/grpc/experimental/stats/instrumentregistry"
+	"google.golang.org/grpc/stats"
+)
+
+// InstrumentDescriptor is a data of a registered instrument (metric).
+type InstrumentDescriptor struct {
```

Comment:
Aren't these things that third-party plugins will eventually need to work with?  (I.e. should be in `experimental`?)

Reply:
Oh hmmmm you're right. So the only things kept internal are what a gRPC component calls into, and this will need to be read by external users in their stats plugins so they can create instruments based off it.

Reply:
I asked about this in the chat; thinking about it do I need to move the whole instrument registry into experimental actually? A user can also write their own balancer/resolver, which should be able to register instruments to eventually record on so I think that all the symbols/logic need to be external (in experimental). What do you think?

Reply:
Discussed in chat with Yash; decided to move this whole instrument registry to experimental and make externally visible.